meson.build

```diff
@@ -0,0 +1,272 @@
+project('libass', 'c', license: 'ISC',
+        meson_version: '>= 0.60.0',
+        # cpp_std specified because meson compiler options are currently global, can be set per-subproject with 0.63
+        default_options: ['c_std=c99', 'cpp_std=c++11', 'buildtype=debugoptimized', 'warning_level=2', 'default_library=static'],
+        version: '0.17.1')
+
+conf = configuration_data()
+deps = []
+test_deps = []
+
+host_system = host_machine.system()
+
+# Compiler setup
+
+cc = meson.get_compiler('c')
+
+if cc.get_argument_syntax() == 'gcc'
+    cflags = [
+        '-Wno-sign-compare',
+        '-Wno-unused-parameter',
+        '-Werror-implicit-function-declaration',
+        '-Wstrict-prototypes',
+        '-Wpointer-arith',
+        '-Wredundant-decls',
+        '-Wno-missing-field-initializers'
+    ]
+    add_project_arguments(cc.get_supported_arguments(cflags), language: 'c')
+endif
+
+test_args = []
+
+if cc.get_id() != 'msvc'
+    test_args += '-D_POSIX_C_SOURCE=200809L'
+    add_project_arguments('-D_POSIX_C_SOURCE=200809L', language: 'c')
+endif
```

Comment:
Question: will the later `has_header_symbol` not already use the project arguments or is there a reason why we don't hand it all current project arguments instead of a `test_args` subset?

Reply:
> but I also have no clue how to determine if meson is actually using the project compiler flags during setup checks

`meson-logs/meson-log.txt` in the build folder shows all the commands run during configuring.

Reply:
Since all relevant checks currently have (potentially less efficient) fallback paths, the build still succeeding isn’t saying much. Checking log of hte newest CI run, indeed it now fails to find e.g. native `str(n)dup`:
```
Checking for function "strdup" : YES 
Header "string.h" has symbol "strdup" : NO 
Checking for function "strndup" : YES 
Header "string.h" has symbol "strndup" : NO 
```

If there’s no convenient way to use proejct args for tests, just restoring the `test_args` duplication is fine (cc @eli-schwartz)

Reply:
Meson can add flags to a global stack with add_project_arguments, or add them to an array and use them precisely where needed. The tricky thing about compiler tests is that adding macros can change the result of the test, so: is that wanted or not? :D If it is unwanted, it's difficult to remove when global stack applies. If it's wanted, it is straightforward to add those args to an array and then pass that array both to `add_project_arguments(test_args, language: 'c')` and to compiler tests that need it.